projects/packages/connection/src/class-plugin-storage.php

```diff
@@ -182,6 +182,10 @@ public static function configure() {
 	public static function update_active_plugins_option() {
 		// Note: Since this options is synced to wpcom, if you change its structure, you have to update the sanitizer at wpcom side.
 		update_option( self::ACTIVE_PLUGINS_OPTION_NAME, self::$plugins );
+
+		if ( ! class_exists( 'Automattic\Jetpack\Sync\Settings' ) || ! \Automattic\Jetpack\Sync\Settings::is_sync_enabled() ) {
```

Comment:
I wonder if we need to handle this option in a similar way as the package versions, where we only update the option if the request to WPCOM was successful (or if it was handled by Sync in this case).

With the current approach, I think we could get stuck with an outdated `jetpack_connection_active_plugins` option value on WPCOM if the request fails. The remote site's option would be correct, so we wouldn't detect that an update is needed.

Reply:
I agree that the priority here is to have always up-to-date list of the connected active plugins in the remote site. Given this and the fact that this is a "failsafe" solution for cases where Sync is not present I'd prefer having the remote site updating the option no matter if the remote request failed.

I'm also currently working on a PR that will early send this option when the site is registered (along with the `jetpack_version`) so I believe we should be good here.

Does that sound agreeable? Very interested on your thoughts :)